src/display/display_utils.js

```diff
@@ -658,11 +658,21 @@ class OutputScale {
    * @returns {boolean} Returns `true` if scaling was limited,
    *   `false` otherwise.
    */
-  limitCanvas(width, height, maxPixels, maxDim) {
+  limitCanvas(width, height, maxPixels, maxDim, capCanvasAreaFactor) {
```

Comment:
Nit: The other parameters don't contain the word "Canvas" to shorten the method signature a little bit.

Also, since the thumbnail rendering also uses this method (where the cap-parameter won't be used) let's give the new parameter a default value.
```suggestion
  limitCanvas(width, height, maxPixels, maxDim, capAreaFactor = -1) {
```